### PR TITLE
filer: reject a proxyChunkId that isn't a well-formed fid

### PR DESCRIPTION
**Pull request by chrislusf**
`LookupFileId` only requires the fid to contain a single comma (`len(strings.Split(fileId, ",")) == 2`), and `proxyToVolumeServer` pastes the value straight into the volume server URL path. So `?proxyChunkId=3,x/../../status` resolves volume 3 and then addresses an endpoint the caller never named.

Verified end to end with a standalone `ServeMux` + `http.DefaultClient` repro:

```
target URL: http://127.0.0.1:57331/3,x/../../status
req.URL.RequestURI() = "/3,x/../../status"
status=200
body=VOLUME STATUS JSON: disk paths, volume inventory
```

Go sends the dot segments verbatim, the volume server's mux cleans the path and 301s, and the filer's client follows the redirect and relays the body. That reaches any handler registered on the volume server — `/status`, `/stats/*`, the UI — through a filer that operators often treat as the only exposed surface, with volume servers on a private network.

This is pre-existing and independent of any JWT handling; it needs no keys configured and works on a plain GET.

## Fix

Parse the fid before the lookup, answer 400 when it doesn't parse. A trailing `_N` delta suffix from batch assigns (`operation/submit.go:83,197`) is legal, so it's stripped first the way the volume server does in `maybeCheckJwtAuthorization`.

Reuses `needle.ParseFileIdFromString` rather than hand-rolling a character filter, so the accepted set is exactly what the rest of the tree considers a fid.

## Tests

Table over the validator covering the canonical form, both delta forms, dot segments with and without a delta suffix, `?`/`#` injection, and truncated input. Plus a handler-level test asserting 400 — it runs with `fs.filer` nil, so reaching the lookup would panic; surviving the call proves the fid is rejected before any lookup happens.

`go build ./weed/...`, `go vet ./weed/server/` and the full `./weed/server/` package pass.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved validation of proxy file identifiers to catch malformed IDs and traversal-style payloads early.
  * Requests with invalid formats, unexpected separators, or unsafe patterns are now rejected with **400 Bad Request** before any volume-server lookup is attempted.
* **Tests**
  * Expanded coverage with additional malformed proxy chunk ID patterns, including a regression test asserting traversal inputs are rejected with **400**.

### CODE REVIEW

File: weed/server/filer_server_handlers_proxy.go [hunk 1]
```diff
@@ -48,8 +50,30 @@ func releaseProxySemaphore(host string) {
...
+// to /status, which the filer follows and relays. A trailing _N delta suffix is
+// legal, and is stripped before parsing the way the volume server does.
+func validateProxyChunkId(fileId string) error {
+	base := fileId
+	if sepIndex := strings.LastIndex(base, "_"); sepIndex > 0 {
+		base = base[:sepIndex]
+	}
+	_, err := needle.ParseFileIdFromString(base)
+	return err
+}
```
[the remark this answers is not in the document]
Comment (chrislusf) [reply weed/server/filer_server_handlers_proxy.go:86]: Confirmed and fixed. You're right, and so is the reasoning about why `maybeCheckJwtAuthorization` gets away with the same pattern — its fid comes out of a path the mux already parsed and so cannot hold a `/`, whereas this one is raw query input.

I reproduced both shapes before fixing:

```
"3,01637037d6_1/../../status"  base="3,01637037d6"  validate=<nil>
"3,01637037d6_../../status"    base="3,01637037d6"  validate=<nil>
```

Rather than returning a separate error from the validator, I put the guard in the strip itself: the suffix is removed only when it is a non-empty run of digits, otherwise the fid is returned whole and `ParseFileIdFromString` rejects it on the `_` or `/`. That keeps one code path and means the same helper stays correct when the stacked PR reuses it for JWT minting.

Added to the table: `_1/../../status`, `_../../status`, `_1/../../stats/counter`, plus `_`, `_abc` and `_1a` for the suffix shape itself. The two traversal payloads also went into the handler-level test, which asserts 400 with a nil `fs.filer` — reaching the lookup panics, so a bypass can't pass silently. Reverting the digit guard now fails both tests.

File: weed/server/filer_server_handlers_proxy.go [hunk 2]
```diff
@@ -48,8 +50,30 @@ func releaseProxySemaphore(host string) {
...
+// LookupFileId only requires a single comma, and the value is pasted into the
+// volume server URL path, so "3,x/../../status" resolves to a volume the caller
+// never named -- the volume server's mux cleans the dot segments and redirects
+// to /status, which the filer follows and relays. A trailing _N delta suffix is
+// legal, and is stripped before parsing the way the volume server does.
+func validateProxyChunkId(fileId string) error {
+	base := fileId
+	if sepIndex := strings.LastIndex(base, "_"); sepIndex > 0 {
+		base = base[:sepIndex]
+	}
```
[the remark this answers is not in the document]
Comment (chrislusf) [reply weed/server/filer_server_handlers_proxy.go]: Confirmed and fixed — thanks, this was a real bypass in the fix.

I took a slightly different shape than the suggested diff: instead of returning a new error from the validator, the digit check lives inside the strip. When the suffix isn't a non-empty run of digits the fid is returned whole, and `ParseFileIdFromString` then rejects it on the `_` or the `/`. Same outcome, one code path, no extra error string — and it keeps the helper correct where the stacked PR reuses it for JWT minting.

Test table now covers `_../../status` (your payload), `_1/../../status`, `_1/../../stats/counter`, and the suffix shapes `_`, `_abc`, `_1a`.

**Comment by chrislusf** [comment]
Pushed the fix for the delta-suffix bypass both reviewers flagged; it was real and I reproduced it before changing anything:

```
"3,01637037d6_1/../../status"  base="3,01637037d6"  validate=<nil>
"3,01637037d6_../../status"    base="3,01637037d6"  validate=<nil>
```

The strip now removes the suffix only when it is a non-empty run of digits; anything else leaves the fid whole so `ParseFileIdFromString` rejects it on the `_` or the `/`. I put the guard in the strip rather than adding a second error path in the validator, so the same helper stays correct where the stacked PR reuses it for JWT minting.

New cases: `_1/../../status`, `_../../status`, `_1/../../stats/counter`, and the suffix shapes `_`, `_abc`, `_1a`. The two traversal payloads also went into the handler-level test. Reverting the digit guard fails both tests — the handler one panics on the nil `fs.filer`, which is precisely the signal that the lookup was reached.

Full `weed/server` suite and vet pass.
